pkg/skaffold/debug/container.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"fmt"
+
+	shell "github.com/kballard/go-shellquote"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug/annotations"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug/types"
+)
+
+// containerTransforms are the set of configured transformers
+var containerTransforms []containerTransformer
+
+// RegisterContainerTransformer allows calling packages to register their own
+// transformer implementation with the global transform list.
+// It returns a function to reset the transform list to its original state,
+// which is useful only for testing.
+func RegisterContainerTransformer(t containerTransformer) func() {
+	cpy := make([]containerTransformer, len(containerTransforms))
+	copy(cpy, containerTransforms)
+	resetFunc := func() { containerTransforms = cpy }
```

Comment:
> otherwise you could toss any transforms that were added after this particular call

that's the point though 😆 this is only useful in tests where we are sequentially adding transforms to the global list, and we need to remove them later. I suppose I could also just add an `UnregisterContainerTransform` function, but it seemed easier just to save the state and then reset it when the test exits.

Reply:
This code isn't in a `_test.go` or behind a test tag, so we need to make it robust.

Reply:
yeah you're right. I changed this to have an `UnregisterContainerTransformer()` counterpart, and I changed the `containerTransformers` list to be a map, which makes it easier to add/remove elements.